### PR TITLE
[SPARK-47066][INFRA][FOLLOW-UP] Deduplicate Apple Silicon Maven build definition

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR is a followup of https://github.com/apache/spark/pull/45126 that deduplicates the build code.

### Why are the changes needed?

For better maintenance of the code

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

This PR should test it out.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -792,14 +792,18 @@ jobs:
...
-        java:
-          - 17
-          - 21
-    runs-on: ubuntu-22.04
+        include:
+          - java: 17
+            os: ubuntu-22.04 
+          - java: 21
+            os: ubuntu-22.04 
+          - java: 17
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:804]:
```suggestion
          - java: 21
```

**Comment by HyukjinKwon** [comment]
@dongjoon-hyun mind taking a look when you find some time?

**Comment by dongjoon-hyun** [comment]
Actually, I didn't chose this way because the pipeline name is `java-other-versions`, @HyukjinKwon .

However, feel free to merge this if you want.

**Review by dongjoon-hyun — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -82,7 +82,7 @@ jobs:
           sparkr=`./dev/is-changed.py -m sparkr`
           tpcds=`./dev/is-changed.py -m sql`
           docker=`./dev/is-changed.py -m docker-integration-tests`
-          # 'build' and 'java-other-versions' are always true for now.
+          # 'build' and 'maven-build' are always true for now.
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:85]: Last commit was just renaming to address a comment. Let me merge this in.

**Pull request closed without merging**